pkg/threescale/backend/backend.go

```diff
@@ -421,19 +419,29 @@ func (b *Backend) GetPeer() string {
 	return b.client.GetPeer()
 }
 
+// calculateDeltas returns a set of metrics that we can report to 3scale using the lowest period known in the counters
+// for metrics with rate limits attached
 // calculateDeltas returns a set of metrics that we can report to 3scale using the lowest period known in the counters
 // for metrics with rate limits attached
 func (a *Application) calculateDeltas() api.Metrics {
 	deltas := make(api.Metrics)
 
-	for metric := range a.LimitCounter {
-		lowestPeriod := a.LimitCounter.getLowestKnownPeriodForMetric(metric)
-		if lowestPeriod != nil {
-			if lastKnownState, ok := a.RemoteState[metric][*lowestPeriod]; ok {
-				toReport := a.LimitCounter[metric][*lowestPeriod].current - lastKnownState.current
-				deltas.Add(metric, toReport)
-			}
+	for metric, counters := range a.LocalState {
+		// we want to handle the lowest granularity only here
+		if canHandle := a.compareLocalToRemoteState(metric, 0); !canHandle {
+			// TODO - add logging
+			continue
 		}
+
+		lowestLocalGranularity := counters[0]
+		lowestRemoteGranularity := a.RemoteState[metric][0]
+
+		if lowestLocalGranularity.PeriodWindow.Period != lowestRemoteGranularity.PeriodWindow.Period {
```

Comment:
Maybe add a comment that we maintain the remote state and anything not in the remote is our delta?